Adding TE branch

KidWizardOfTheWeb opened the pull request:
Followed the steps in order to add a new patcher. It seems there is more required to do this, but I'm submitting a PR at this point to handle it first.

KidWizardOfTheWeb committed:
Adding TE branch

Sewer56 commented on Sewer56.Patcher.Regravitified.Te/Utility/Xor.cs:
```diff
@@ -0,0 +1,162 @@
+/// <summary>
+/// Implementation of XOR over a 32-byte array.
+/// </summary>
+public static class Xor
```
The SIMD'ed XOR and Compression routines were used originally by the self contained (a.k.a. fake) patcher. Pretty sure this would be dead code right now.

KidWizardOfTheWeb replied:
Was a copy-over from the folder, can remove easily. Do you suggest removing the Utility folder as a whole as well?

Sewer56 commented on Sewer56.Patcher.Regravitified.Te/Utility/Compression.cs:
```diff
@@ -0,0 +1,54 @@
+﻿#if SRDXSelfContained
```
This is currently dead code.

KidWizardOfTheWeb replied:
Same as above, will remove.

Sewer56 commented on Sewer56.Patcher.Regravitified.Te/Theme/Te/Images/NoType3.png:
WITH SCARF!!!1!1

(This image appears to be currently unused)

KidWizardOfTheWeb replied:
Yeah just needed a placeholder. Will get actual images for the merge.

Sewer56 commented:
Otherwise lools good to me.

KidWizardOfTheWeb committed:
Fixed PR issues to prepare for merge into main

KidWizardOfTheWeb commented:
Fixed all the image issues and removed dead code.
In order to actually include this into the CLI portion/building, it seems that code needs to be added to the CLI project in order to handle actually generating the patcher. Is this something that I need to handle or do you take care of that portion?

Sewer56 commented:
> Fixed all the image issues and removed dead code. In order to actually include this into the CLI portion/building, it seems that code needs to be added to the CLI project in order to handle actually generating the patcher. Is this something that I need to handle or do you take care of that portion?

Not at my desk right now, but check the `#define` statement, probably gotta add in a statement somewhere along the rest.

KidWizardOfTheWeb commented:
![image](https://github.com/user-attachments/assets/c0e8b118-9a3b-463e-bac6-65fa034e864a)
Ah I see, probably here then. Then just `dotnet restore` and it should be up I'd think.
Just didn't want to mess with source without permission.

KidWizardOfTheWeb committed:
TE patcher should be fully enabled now, needs testing to make sure.

KidWizardOfTheWeb commented:
I'll test this in a bit to make sure everything works properly, then after that start working on the network/server integration in a new PR.

Sewer56 commented on Sewer56.Patcher.Regravitified.Cli/Commandline.cs:
```diff
@@ -10,7 +10,7 @@
 using Sewer56.Patcher.Riders.Common.Utility;
 #endif
-#if REGRAV
+#if REGRAV || SRTE
```
Doublecheck if this define is needed. Might not be needed and throw a warning during build.

KidWizardOfTheWeb replied:
I've included it there because it throws an error due to not having the NKIT utility in the path. But that is due to the lower comment, so I will remove it now.

Sewer56 commented on Directory.Build.props:
```diff
@@ -16,7 +16,7 @@
           DEV # Enables some developer tools.
     -->
-    <DefineConstants>$(DefineConstants);SRDX</DefineConstants>
+    <DefineConstants>$(DefineConstants);SRTE</DefineConstants>
```
Don't change the default, because it will make the result of default build inconsistent with the README. Instead update the README to add `SRTE` as one of the possibilities there.

KidWizardOfTheWeb commented:
I have made the changes, but testing the SRTE mode does not work. Publishing seems to happen fine, but the app will not open at all. Will post current commit for review.

KidWizardOfTheWeb committed:
Updated readme and removed include in Commandline.cs

KidWizardOfTheWeb committed:
Fixed a simple icon define

Sewer56 commented:
Want me to test in Windows? I can reboot moment I finish up work.
IIRC publish sometimes gets borked, and is missing an icon.

KidWizardOfTheWeb commented:
Yes please.
~~Weirdly enough the icon is actually DX's icon, even though it's not in the Images folder for TE. I just fixed the pathing too and same issue.~~
Correction: Had to restart file explorer process for the icon to work. But the program remains unable to open.
Also please check the new comment on commandline.cs for the issue I had earlier.

KidWizardOfTheWeb committed:
Slight adjustment to README.md

KidWizardOfTheWeb commented:
![image](https://github.com/user-attachments/assets/c17fd0e8-9193-443a-b379-c1ce6a643961)
I tried to do what debugging I could, and it seems that the program in SRTE mode fails on this step. I'm not sure why it cannot initialize components, there must be something missing.

KidWizardOfTheWeb committed:
TE is able to build properly now and run GUI window

KidWizardOfTheWeb committed:
Extra file adjustments for compat w/ TE online patcher

KidWizardOfTheWeb committed:
Reverted main program runtime method to suit the next patcher update
